test/integration/targets/pids/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+# Test code for the pid module
+# Copyright 2018, Saranya sridharan
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: "Installing PidOf"
+  yum:
+    name: sysvinit-tools
+    state: present
+  when: ansible_distribution == 'CentOS'
+
+  
+- name:  "Installing PidOf in freebsd"  
+  pkgng:
+    name: pidof
+    state: present
+  when: ansible_distribution == 'FreeBSD'
+  
+- name: Install  PidOf in (OSX)
+  block:
+    - name: Find brew binary
+      command: which brew
+      register: brew_which
+    - name: Get owner of brew binary
+      stat: path="{{ brew_which.stdout }}"
+      register: brew_stat
+    - name: "Install package"
+      homebrew:
+        name: pidof
+        state: present
+        update_homebrew: no
+      become: yes
+      become_user: "{{ brew_stat.stat.pw_name }}"
+      
+  when: ansible_os_family == 'Darwin'
+
+- name: "Installing setproctitle"
+  pip:
+    name: setproctitle
+    version: 1.1.10
+
+- name: "Checking empty result" 
+  pids: 
+    name: "blahblah"
+  register: emptypids
+  
+   
+- name: "Verify the Process IDs (PIDs) returned is empty"
+  assert:
+    that:
+    - emptypids is not changed
+    - emptypids.pids == [] 
+   
+
+- name: Executing the python script
+  command: ./sample.py
+  args:
+    chdir: /var/root/ansible/test/integration/targets/pids/tasks
+  async: 100
+  poll: 5
+  when: ansible_os_family == 'Darwin'
+  
+  
+- name: Executing the python script
+  command: ./sample.py
+  args:
+    chdir: /root/ansible/test/integration/targets/pids/tasks
+  async: 100
+  poll: 5
+  when: ansible_os_family != 'Darwin'
+  
+- name: "Getting the process IDs (PIDs) of checkobtainedpid module through linux command"
+  command: "pidof checkobtainedpid"
+  register: pidsobtained
+  failed_when: no
+      
+- name: "Checking the process IDs (PIDs) of checkobtainedpid module" 
+  pids: 
+    name: "checkobtainedpid"
+  register: pids
+  
+- name: "Verify the Process IDs (PIDs) returned is equal to the actual process IDs (PIDs) obtained"
+  assert:
+    that:
+    - "pids.pids | join(' ') == pidsobtained.stdout"
```

Comment:
Don't you also want to assert that there is at least one PID?

Reply:
Is it needed @felixfontein ? Isn't the above condition check our requirement ?

Reply:
Wasn't the original motivation to have one test of the module where it returns a nontrivial result? If that is the case (otherwise: why go through all that trouble and install `setproctitle`) you really need this to make sure this actually happens. That this fails indicates that your test is **not** doing what you want it to do, so instead of removing this part of the test, you should fix it.

Reply:
I don't think your latest commit will fix this. The problem is that `checkobtainedpid` seems to be no longer running when the `pids` module and the `pidof` command are run.

Reply:
Have you actually tried to run this locally (on a *nix system): execute `sample.py` in one shell, and running `pidof checkobtainedpid` in another shell? Did this actually return something?

Also note that your PR includes `sample.py` *twice*. I guess you only need one of the two copies.

Reply:
I have used some other variable{{ ansible_facts.env['PWD'] }} to get pwd . There was some issue with setproctitle installation earlier, I was able to get the pid of sample.py in the defined name after installating gcc and python-devel in my environment. But here it still fails